arviz/tests/external_tests/test_data_pyjags.py

```diff
@@ -99,3 +138,55 @@ def test_inference_data_attrs(posterior, prior, save_warmup, warmup_iterations:
     fails = check_multiple_attrs(test_dict, arviz_inference_data_from_pyjags_samples_dict)
     print(fails)
     assert not fails
+
+
+@pytest.fixture()
+def jags_prior_model() -> pyjags.Model:
+    return pyjags.Model(
+        code=EIGHT_SCHOOL_PRIOR_MODEL_CODE, data={"J": 8}, chains=4, threads=4, chains_per_thread=1
+    )
+
+
+@pytest.fixture()
+def jags_posterior_model() -> pyjags.Model:
+    return pyjags.Model(
+        code=EIGHT_SCHOOL_POSTERIOR_MODEL_CODE,
+        data=EIGHT_SCHOOL_DATA,
+        chains=4,
+        threads=4,
+        chains_per_thread=1,
+    )
+
+
+@pytest.fixture()
+def jags_prior_samples(jags_prior_model: pyjags.Model) -> tp.Dict[str, np.ndarray]:
+    return jags_prior_model.sample(
+        NUMBER_OF_WARMUP_SAMPLES + NUMBER_OF_POST_WARMUP_SAMPLES, vars=PARAMETERS
+    )
+
+
+@pytest.fixture()
+def jags_posterior_samples(jags_posterior_model: pyjags.Model) -> tp.Dict[str, np.ndarray]:
+    return jags_posterior_model.sample(
+        NUMBER_OF_WARMUP_SAMPLES + NUMBER_OF_POST_WARMUP_SAMPLES, vars=VARIABLES
+    )
+
+
+@pytest.fixture()
+def pyjags_data(
+    jags_prior_samples: tp.Dict[str, np.ndarray], jags_posterior_samples: tp.Dict[str, np.ndarray]
+) -> az.InferenceData:
+    return az.from_pyjags(
+        posterior=jags_posterior_samples,
+        prior=jags_prior_samples,
+        log_likelihood_name="log_like",
+        save_warmup=True,
+        warmup_iterations=NUMBER_OF_WARMUP_SAMPLES,
+    )
+
+
+def test_waic(pyjags_data: az.InferenceData):
+    waic = az.waic(pyjags_data)
+
+    assert -31.0 < waic.waic < -30.0
+    assert 0.75 < waic.p_waic < 0.90
```

Comment:
I think we should test groups and variables presence using `check_multiple_attrs` instead of calling waic. If necessary there can be more assertions in addition to `check_multiple_attrs` to check correct dims, shapes...

Reply:
I added a check for the presence of the log-likelihood via `check_multiple_attrs`. I think that it does not hurt (and is probably a good idea) to check that what is in the likelihood attribute is actually the likelihood and not some other data grabbed from the posterior. Please let me know if this is ok for you or you would prefer to have it removed.

Reply:
I added a check for the presence of the log-likelihood via `check_multiple_attrs`. I think that it does not hurt (and is probably a good idea) to check that what is in the likelihood attribute is actually the likelihood and not some other data grabbed from the posterior. Please let me know if this is ok for you or you would prefer to have it removed.